File: H-Index_II.java
/**
 * Problem: https://leetcode.com/problems/h-index-ii/
 */

public class Solution {
    public int hIndex(int[] citations) {
        if(citations.length == 0) return 0;
        int left = 0;
        int right = citations.length - 1;
        while(left <= right) {
            int mid = (left + right) / 2;
            if(citations[mid] >= citations.length - mid) right =mid-1;
            else left = mid + 1;
        }
        return citations.length - left;
    }
}

File: Swift/h-index-ii.swift
/**
 * https://leetcode.com/problems/h-index-ii/
 * 
 * 
 */ 
// Date: Thu Jun 18 09:09:00 PDT 2020
class Solution {
    func hIndex(_ citations: [Int]) -> Int {
        var left = 0
        var right = citations.count
        while left < right {
            let mid = left + (right - left) / 2
            if citations[mid] < (citations.count - mid) {
                left = mid + 1
            } else {
                right = mid
            }
        }
        return citations.count - left
    }
}
